src/main/java/org/eclipsefoundation/marketplace/helper/JavaVersionHelper.java

```diff
@@ -0,0 +1,71 @@
+/* Copyright (c) 2019 Eclipse Foundation and others.
+ * This program and the accompanying materials are made available
+ * under the terms of the Eclipse Public License 2.0
+ * which is available at http://www.eclipse.org/legal/epl-v20.html,
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipsefoundation.marketplace.helper;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Used to handle the conversion of Java versions to and from storage mediums.
+ * 
+ * @author Martin Lowe
+ *
+ */
+public class JavaVersionHelper {
+	private static final Pattern MAJOR_VERSION_CHECK = Pattern.compile("^\\d\\.([^\\.\\D]+)");
+
+	/**
+	 * Converts value to strip out parts of Java version string that cause issues
+	 * with natural number sorting in databases.
+	 * 
+	 * @param val the java version string to convert (e.g. 1.7, 1.10)
+	 * @return modified and more sortable version string
+	 */
+	public static final String convertToDBSafe(String val) {
+		String out = val;
+		if (out == null) {
+			return null;
+		}
+		// strip out the major Java version as it isn't useful.
+		Matcher m = MAJOR_VERSION_CHECK.matcher(out);
+		if (m.matches()) {
+			out = m.group(1);
+		}
+		// if there is still a dot version separator, strip all values after and
+		// including the dot
+		int idx = out.indexOf('.');
+		if (idx > -1) {
+			out = out.substring(0, idx);
+		}
+
+		return out;
+	}
+
+	/**
+	 * Converts the sortable DB value for the Java version to something more in line
+	 * with what is expected by the user.
+	 * 
+	 * @param val the sortable stored version string
+	 * @return display value of the sortable Java version
+	 */
+	public static final String convertToDisplayValue(String val) {
```

Comment:
Add some unit tests as well. I'm pretty sure we don't want to display 1.11 for Java 11 and AFAICT, this code does exactly that.